src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_INCREASE_NUMBER = 2;
+    private static final int FIRST_PRIME_NUMBER = 31;
+    private static final int SECOND_PRIME_NUMBER = 17;
+
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int bucketPosition = getBucketPosition(key);
+        Node<K, V> previousNode = null;
+        Node<K, V> bucketNode = table[bucketPosition];
+        while (bucketNode != null) {
+            if (Objects.equals(key, bucketNode.key)) {
+                bucketNode.value = value;
+                return;
+            }
+            previousNode = bucketNode;
+            bucketNode = bucketNode.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (previousNode != null) {
+            previousNode.next = newNode;
+        } else {
+            table[bucketPosition] = newNode;
+        }
```

Comment:
you can check is bucketNode null, and if yes put new node to the bucket and if not, else - create new additional variable `previousNode` to go through all elements inside bucket. that will improve you logic 
```
if (bucketNode == null) {
  table[bucketPosition] = newNode;
} else {
  Node<K, V> previousNode = null;
  while().....
}
```